src/easypost.js

```diff
@@ -221,16 +232,47 @@ export default class EasyPostClient {
       request.auth(this.key);
     }
 
-    if (params !== {} && params !== undefined) {
+    // it would be ideal if this "full url with params" could be gotten from superagent directly,
+    // but it doesn't seem to be possible
+    const url = new URL(urlPath);
+
+    if (params !== undefined) {
       if (method === METHODS.GET || method === METHODS.DELETE) {
         request.query(params);
+        Object.entries(params).forEach(([key, value]) => {
+          url.searchParams.append(key, value);
+        });
       } else {
         request.send(params);
       }
     }
 
+    const baseHooksValue = {
+      method,
+      path: url.toString(),
+      requestBody: request._data,
+      requestHeaders,
```

Comment:
```suggestion
      headers,
```

This should be called `headers` to follow our tech design

Reply:
Hmmm... do you think it might be a little bit confusing to have both `headers` and `responseHeaders` in the same object on the response? I renamed it to `requestHeaders` to help distinguish between the two.

Reply:
We should rename it to stay consistent across all CLs. The `responseHook` should not be getting `requestHeader`, you will need to make some changes in line 267 `...baseHooksValue`

Reply:
done